code/modules/mob/living/carbon/superior_animal/roach/types/kaiser.dm

```diff
@@ -0,0 +1,131 @@
+/*
+Boss of this maints.
+Has ability of every roach.
+*/
+
+/mob/living/carbon/superior_animal/roach/kaiser
+	name = "Kaiser Roach"
+	desc = "A glorious emperor of roaches."
+	icon = 'icons/mob/64x64.dmi'
+	icon_state = "kaiser_roach"
+	density = TRUE
+
+	turns_per_move = 4
+	maxHealth = 2000
+	health = 2000
+	contaminant_immunity = TRUE
+
+	var/datum/reagents/gas_sac
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	move_to_delay = 8
+	mob_size =  3  // The same as Hivemind Tyrant
+
+	var/distress_call_stage = 3
+
+	var/health_marker_1 = 1500
+	var/health_marker_2 = 1000
+	var/health_marker_3 = 500
+
+	blattedin_revives_left = 0
+
+	// TODO: Add a special type of meat for Kaiser
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/roachmeat/fuhrer
+	meat_amount = 15
+	sanity_damage = 3
+
+/mob/living/carbon/superior_animal/roach/kaiser/New()
+	..()
+	gas_sac = new /datum/reagents(100, src)
+	pixel_x = -16  // For some reason it doesn't work when I overload them in class definition, so here it is.
+	pixel_y = -16
+
+
+/mob/living/carbon/superior_animal/roach/kaiser/Life()
+	. = ..()
+	if(stat != CONSCIOUS)
+		return
+
+	if(stat != AI_inactive)
+		return
+	
+	if(can_call_reinforcements())
+		distress_call()
+
+	gas_sac.add_reagent("blattedin", 1)
+	if(prob(7))
+		gas_attack()
```

Comment:
Making a gas attack without anything to attack seems rather pointless

Reply:
This reagent should heal roaches.